### PR TITLE
ENH: Clarify error message when reindexing fails (#42000)

### LINKED ISSUE #42000: ENH: Clarify error message when reindexing on an axis containing duplicates
#### Is your feature request related to a problem?

Let's consider the following snippet:

```python
pd.DataFrame(index=[pd.to_datetime("2019"), pd.to_datetime("2019"), pd.to_datetime("2020")]).asfreq("M")
```

This will raise an exception as the to be resampled dataframe has an axis with duplicates. The problem: I stumpbled at least twice on this problem and never understood what the error message wanted to tell me on first sight.

When seeing just the message (wihtout looking at the backtrace) I always thought this is about providing duplicate axis names or something like this.

```python
ValueError: cannot reindex from a duplicate axis
```

#### Describe the solution you'd like

Maybe the exception message could be clarified. The comment two lines above the exception always saves me:

```python
# trying to reindex on an axis with duplicates
```

So maybe  something like this could help?

```python
raise ValueError("Cannot reindex on an axis with duplicate values")
```

#### API breaking implications

None I guess.

The message would additionally need to be changed in some tests and in the docs.

#### Additional context

If there is interest I could draft a PR.

### PR DESCRIPTION
**Pull request by aberres**
- [x] closes #42000
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

**Review by phofl — APPROVED** [review]
lgtm

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -3708,7 +3708,7 @@ def _validate_can_reindex(self, indexer: np.ndarray) -> None:
         """
         # trying to reindex on an axis with duplicates
         if not self._index_as_unique and len(indexer):
-            raise ValueError("cannot reindex from a duplicate axis")
+            raise ValueError("Cannot reindex on an axis with duplicate labels")
```
Comment (simonjayhawkins) [inline pandas/core/indexes/base.py:3711]: I _think_ it's common practice not to use capitalization

SyntaxError: invalid syntax
IndexError: list index out of range
IndentationError: unexpected indent
IndentationError: expected an indented block

Comment (jreback) [reply pandas/core/indexes/base.py:3711]: agree. @aberres can you update & ping on green.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by aberres** [comment]
@jreback Updated. Not green, but the failing check seems unrelated?

**Comment by simonjayhawkins** [comment]
> @jreback Updated. Not green, but the failing check seems unrelated?

should now be fixed

**Comment by simonjayhawkins** [comment]
i'll merge with a 1.4 milestone. but a trivial change so could backport.

**Comment by simonjayhawkins** [comment]
Thanks @aberres

**Pull request merged**